### PR TITLE
Workers now build forts

### PR DESCRIPTION
**Pull request by tuvus**
This PR works on adding back in fort building AI for workers. I will probably post some other questions regarding the desired behavior.

Is the picture below unreasonable for the fort-building AI? The AI has plenty of workers because I changed their priority and forts have no maintenance cost. They also aren't losing stats for building the forts on improbable tiles.

<details><summary>Picture</summary>

AIFortBuilding

</details>

Is it really better for a fort to be built on an owned tile vs a neutral tile? Is it bad for a fort to be built adjacent to a city?

FYI, we are only looking at fort building here and whether or not the AI will use the forts. That's a different question entirely.

### CODE REVIEW

**Comment by yairm210** [comment]
I don't know what should be considered OK fort building practices 🤔 
I think we need players to say

**Comment by tuvus** [comment]
Okay, I will ask discord then.

File: core/src/com/unciv/logic/automation/unit/UnitAutomation.kt [hunk 1]
```diff
@@ -369,8 +369,8 @@ object UnitAutomation {
         val tilesThatCanWalkToAndThenPillage = unitDistanceToTiles
             .filter { it.value.totalDistance < unit.currentMovement }.keys
             .filter { unit.movement.canMoveTo(it) && UnitActionsPillage.canPillage(unit, it)
-                    && (it.canPillageTileImprovement()
-                    || (it.canPillageRoad() && it.getRoadOwner() != null && unit.civ.isAtWarWith(it.getRoadOwner()!!)))}
+                    && ((it.canPillageTileImprovement() && !it.getTileImprovement()!!.hasUnique(UniqueType.DefensiveBonus))
```
Comment (SeventhM) [inline core/src/com/unciv/logic/automation/unit/UnitAutomation.kt:372]: Units probably should still pillage stat providing forts and/or forts that that damages them

**Comment by tuvus** [comment]
The code is not perfect. However, the AI will build forts and doesn't seem to build too many of them. Currently, they will only build forts in their own land. Like all things, this will take some more tweaking some other time. Frankly the workers building forts is only a visual effect for the players right now since the unit AI doesn't use them enough. Therefore its probably better to work on some other area.

**Pull request merged**